### PR TITLE
added code for ds_format

### PR DESCRIPTION
**Pull request by krishnabhupatiraju**
@mistercrunch @artwr @jeremyclover Added a macro called ds_format so we can format ds to a different format.

### CODE REVIEW

File: airflow/macros/__init__.py [hunk 1]
```diff
@@ -24,3 +24,22 @@ def ds_add(ds, days):
...
+    as specified in the output format
+
+    :param ds: input string which contains a date
+    :type ds: str
+    :param input_format: input string format. E.g. %Y-%m-%d
+    :type input_format: str
+    :param output_format: output string format  E.g. %Y-%m-%d
+    :type output_format: str
+
+    >>> ds_add('2015-01-01', 5)
```
Comment (mistercrunch) [inline airflow/macros/__init__.py]: Doctests point to different function...

**Comment by mistercrunch** [comment]
also note that you can use `{{ macros.datetime }}` anywhere in Airflow. Note that it's a shortcut to `datetime.datetime`, not `datetime`

**Comment by mistercrunch** [comment]
Feel free to merge once you update the doctests, though this would require a push of code to the workers where macros.datetime won't

**Comment by krishnabhupatiraju** [comment]
Thanks Max. Merged it so it can ship next time. I will use the  macros.datetime in the meantime.

**Pull request merged**